include/zephyr/bluetooth/gatt.h

```diff
@@ -72,6 +76,32 @@ enum {
 	 *  passed to write callback.
 	 */
 	BT_GATT_PERM_PREPARE_WRITE = BIT(6),
+
+	/** @brief Attribute write permission with 128bit entropy key connection.
+	 *
+	 *  If set, write access requires connection using an at least 128bit entropy
+	 *  key derived from one of the following sources:
+	 *  - LE Secure Connections
+	 *  - OOB method
+	 *
+	 *  This permission is a superset of encryption, meaning if set
+	 *  BT_GATT_PERM_READ_ENCRYPT is implied, and can be omitted. It does
+	 *  however not imply BT_GATT_PERM_READ_AUTHEN.
+	 */
+	BT_GATT_PERM_READ_128B_ENTROPY = BIT(7),
```

Comment:
I think this has been covered in the above threads :) 

This PR originally just added a `BT_GATT_PERM_SC`, but since we actually wanted to check for 128-bit entropy (for LE Audio), we have since modified it. 

If we add `BT_GATT_PERM_SC` then we won't be able to support OOB based pairing for LE Audio services, as `BT_GATT_PERM` doesn't work as a "or", but rather an "and" (i.e. we can't just add a `BT_GATT_PERM_OOB` later, as `BT_GATT_PERM_SC | BT_GATT_PERM_OOB` would mean `LESC AND OOB`). 

Furthermore, we may want separate read and write permissions (thus `BT_GATT_PERM_READ_ENCRYPT | BT_GATT_PERM_SC` isn't a good idea); e.g. for CCCDs.

Reply:
Fair enough